### ISSUE TITLE
Branches not being synced to JIRA

### ISSUE THREAD

**Issue by tfmf** [atlassian/github-for-jira#672]
Right now unless you name you branch with the issue name it won't show in the development pane even if there are commits with the ticket name, though the commits will show. However if you force a sync the branch will show up, is this a normal behaviour? 

Steps to reproduce:

1. Create a issue in jira (e.g.: **JIRA-1**)
2. Create a branch in your repo (`my_feature`)
3. Change the **README.md** and commit message with "**JIRA-1 update README**"
4. The the commit will show in **JIRA-1** issue in Jira but not the branch `my_feature`.
5. Go to the My Applications -> Github and force a sync
6. The branch `my_feature` now shows up

**Comment by mboudreau** [comment]
@tfmf That's correct, commits under a branch doesn't automatically link that branch to the ticket.  I think we'll need to revisit this - as well as many other suggestions - but I think originally it was left as is specifically to give the users more control as to what they want associated with the ticket.  

Since we've recently just got full ownership of the app, we'll be reviewing many of these features and implementing them.  My question to you is, how do you think this particular feature would work?  Any ticket ID in any commits in the branch automatically links the branch to the ticket?  What if it was a passing comment about another piece of work?

**Comment by tfmf** [comment]
thanks for the reply @mboudreau. I understand what you're saying but if that's already being done with the force sync from the "manage app" screen I guess it should be the default behaviour. Or at least have an option to set that as a configuration maybe?

**Comment by mboudreau** [comment]
That's a good point. Let me talk to the team. We have a lot to review to try to make a cohesive experience.

**Comment by rachellerathbone** [comment]
Hey @tfmf. Both this and a related issue (#566) have been added to our roadmap. We'll update you again when we can.

**Comment by jmichalicek** [comment]
Looks like this has been on the roadmap for a long time. I was just looking into this because it happens even if the branch itself is named for the issue, if there is also a commit with the issue name, then the branch is not linked.

So branch `JIRA-1` and a commit with `JIRA-1`, the commit seems to be overriding the behavior for having the branch with the name.
